File: Algorithm/075_Sort_Color.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        left is the rightmost boundary of 0
        right is leftmost boundary of 2
        swap current with left/right regard of it's value
        """
        left = current = 0
        right = len(nums) - 1
        
        while current <= right:
            if nums[current] == 0:
                nums[left], nums[current] = nums[current], nums[left]
                left += 1
                current += 1
            elif nums[current] == 2:
                nums[current], nums[right] = nums[right], nums[current]
                right -= 1
            else:
                current += 1

File: Algorithm/075_Sort_Color.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int left = 0;
        int current = 0;
        int right = nums.size() - 1;
        
        while (current <= right) {
            if (nums[current] == 0){
                swap(nums[current], nums[left]);
                left ++;
                current ++;
            }
            else if (nums[current] == 2){
                swap(nums[current], nums[right]);
                right --;
            }
            else{
                current ++;
            }
        }
    }
};